Effisense/sensor-dashboard
# Create backend endpoints and fetch data from database

## Changelog

- Extend the `user` in context to include `organizationId`. Hence, we now call this object `auth`. We can now fetch `user` and `organizationId` from `ctx.auth`.
- Setup basic CRUD operations for `sensor` in `routers/sensor`
- Setup basic CRUD operations for `containerType` in `routers/containerType`
- Move `kysely` dependency to `packages/api` instead of `apps/web`, as this more correctly reflect the area of responsibility. Kysely is used to fetch data from the timeseries database, and should thus not be in `web`.
- Update Prisma schema by removing `Container` entity. I realized that we don't need that relationship at all, and the relationship between `Sensor` and `ContainerType` is sufficient for our queries.
- Setup integration with the SPAN API. As far as I know, we'll only use this the check if the (QR-code-)scanned sensor belongs to a `collectionId` and truly exists in Effisense's `lab5e` instance. This will simply get called when attempting to create a sensor

## How to test

- None of this is easily testable before we have implemented the views to use these.
- Take a look at the code, and try to read out loud what it does. If you have any questions, fire away in the review.
- We'll update this logic when the view are to be implemented. This is more like the backend skeleton.
- Ensure you have the `SPAN_API_TOKEN` in your `.env`. This is available on Vault.
